stl/inc/forward_list

```diff
@@ -411,71 +411,101 @@ public:
 };
 
 template <class _Alnode>
-struct _Flist_insert_after_op {
+struct _Flist_insert_after_op2 {
     // forward_list insert-after operation which maintains exception safety
     using _Alnode_traits = allocator_traits<_Alnode>;
     using pointer        = typename _Alnode_traits::pointer;
     using value_type     = typename _Alnode_traits::value_type;
 
-    explicit _Flist_insert_after_op(_Alnode& _Al_) : _Al(_Al_), _Tail(_STD addressof(_Base)) {}
+    explicit _Flist_insert_after_op2(_Alnode& _Al_) : _Al(_Al_), _Tail(), _Head() {}
 
-    _Flist_insert_after_op(const _Flist_insert_after_op&) = delete;
-    _Flist_insert_after_op& operator=(const _Flist_insert_after_op&) = delete;
+    _Flist_insert_after_op2(const _Flist_insert_after_op2&) = delete;
+    _Flist_insert_after_op2& operator=(const _Flist_insert_after_op2&) = delete;
 
     template <class... _CArgT>
-    pointer _Append_n(typename _Alnode_traits::size_type _Count, const _CArgT&... _Carg) {
+    void _Append_n(typename _Alnode_traits::size_type _Count, const _CArgT&... _Carg) {
         // Append _Count copies of T, constructed from _Carg, to this insert-after operation
+        if (_Count <= 0) {
+            return;
+        }
+
         _Alloc_construct_ptr<_Alnode> _Newnode(_Al);
+        if (_Tail == pointer{}) {
+            _Newnode._Allocate(); // throws
+            _Alnode_traits::construct(_Al, _STD addressof(_Newnode._Ptr->_Myval), _Carg...); // throws
+            _Head = _Newnode._Ptr;
+            _Tail = _Newnode._Ptr;
+            --_Count;
+        }
+
         for (; 0 < _Count; --_Count) {
             _Newnode._Allocate(); // throws
             _Alnode_traits::construct(_Al, _STD addressof(_Newnode._Ptr->_Myval), _Carg...); // throws
-            _Construct_in_place(*_Tail, _Newnode._Ptr); // nothrow
-            _Tail = _STD addressof(_Newnode._Ptr->_Next);
+            _Construct_in_place(_Tail->_Next, _Newnode._Ptr);
+            _Tail = _Newnode._Ptr;
         }
 
-        return _Newnode._Release();
+        (void) _Newnode._Release();
     }
 
     template <class _InIt, class _Sentinel>
-    pointer _Append_range_unchecked(_InIt _First, const _Sentinel _Last) {
+    void _Append_range_unchecked(_InIt _First, const _Sentinel _Last) {
         // Append the values in [_First, _Last) to this insert-after operation
+        if (_First == _Last) { // throws
+            return;
+        }
+
         _Alloc_construct_ptr<_Alnode> _Newnode(_Al);
-        for (; _First != _Last; ++_First) {
+        if (_Tail == pointer{}) {
             _Newnode._Allocate(); // throws
             _Alnode_traits::construct(_Al, _STD addressof(_Newnode._Ptr->_Myval), *_First); // throws
-            _Construct_in_place(*_Tail, _Newnode._Ptr); // nothrow
-            _Tail = _STD addressof(_Newnode._Ptr->_Next);
+            const auto _Newhead = _Newnode._Release();

```

Comment:
There is an implementation divergence between `_Append_n` and this function in that additional temporary. I could not find the reason for that.

Reply:
--_Count can't throw, so we can avoid a bunch of nullptr assignments, but ++_First can throw, so we need to deactivate the `_Alloc_construct_ptr` after ownership of each element has been transferred to `*this`.

Reply:
Thanks for the clarification. We need a “that is correct and I am deeply horrified“ emoji

Reply:
The eyes are close enough right?

Reply:
Nah, depending on the browser they can look cute. :scream: seems appropriate